### ISSUE TITLE
Handling confidential query results

### ISSUE THREAD

**Issue by larspett** [rekonstrukt/swedishbutterflies#11]
When adding fields for filtering to the sql queries, there is a possible problem involving the handling of confidential data. There are data sources (SLU, County administrative boards) and other restricted data categories (secret sites, threatened species) that might be useful to work with but that are currently filtered out. If I add those specifications/columns then it opens the possibility that restricted data gets public. How should we handle this? Outside the normal functions as separate queries and then fed as a separate data source to the function?

**Comment by rekonstrukt** [comment]
Yes, I think so. I guess we'd like to do it similar to how the API does it, generally, if possible.For functions exposing data that is non-public for example using queries that could return sensitive data (or do CRUD operations that change, remove, update or delete data), functions may need to accept non-optional uid and password parameters that the query (or prepared statement) in postgres uses to figure out what to return - unless this can be managed through user roles in postgres. 

One suggestion is to start with one concrete example. Functions that move into this area currently in data.R are sebms_per_update_modified (because it would update a modified time based on user identity) and sebms_users (because it enumerates users, but note that it excludes the password column intentionally).

In general, a shiny user - using an app - when we get there - would be more restricted -  there we'd make sure to pick up the uid and password and use that where needed - while the package user would theoretically be able to do all things that the config.yml-configured db user is capable of given its roles/permissions in postgres.
